app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -98,8 +119,32 @@ open class ProfilesViewModel(
           _profiles.value = profileList
           _filteredProfiles.value = filteredProfiles
           _loading.value = false
+          _isConnected.value = true
+          waitingDone.value = false
+          isWaiting.value = false
+          Log.e(
+              "ConnectionCheck",
+              "Reset flags: waiting=${isWaiting.value}, done=${waitingDone.value}")
         },
-        onFailure = { e -> handleError(e) })
+        onFailure = { e ->
+          Log.e("ConnectionCheck", "Request FAILED")
+          Log.e(
+              "ConnectionCheck",
+              "Current state: waiting=${isWaiting.value}, done=${waitingDone.value}")
+          handleError(e)
+          if (e is com.google.firebase.firestore.FirebaseFirestoreException) {
```

Comment:
Separation of concerns: Ideally, this should be handled in the Firestore Repository, not in the ViewModel.